includes/class-wsuwp-people-directory-roles.php

```diff
@@ -0,0 +1,343 @@
+<?php
+
+class WSUWP_People_Directory_Roles {
+	/**
+	 * @since 0.1.0
+	 *
+	 * @var WSUWP_People_Directory_Roles
+	 */
+	private static $instance;
+
+	/**
+	 * @since 0.1.0
+	 *
+	 * @var array Names for custom VALS roles.
+	 */
+	private static $roles = array(
+		'owner' => 'wsuwp_people_profile_owner',
+		'unit_admin' => 'wsuwp_people_unit_admin',
+	);
+
+	/**
+	 * Maintain and return the one instance and initiate hooks when
+	 * called the first time.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return \WSUWP_People_Directory_Roles
+	 */
+	public static function get_instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new WSUWP_People_Directory_Roles();
+			self::$instance->setup_hooks();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Adds the hooks used to create and manage roles and capabilities.
+	 *
+	 * @since 0.1.0
+	 */
+	public function setup_hooks() {
+		add_action( 'after_switch_theme', array( $this, 'add_roles' ) );
+		add_action( 'switch_theme', array( $this, 'remove_roles' ) );
+		add_action( 'init', array( $this, 'map_role_capabilities' ), 12 );
+		add_action( 'init', array( $this, 'register_wsu_orgs_for_users' ) );
+		add_action( 'personal_options', array( $this, 'extend_user_profile' ) );
+		add_action( 'personal_options_update', array( $this, 'save_user_organization' ) );
+		add_action( 'edit_user_profile_update', array( $this, 'save_user_organization' ) );
+		add_filter( 'user_has_cap', array( $this, 'unit_administration' ), 10, 4 );
+		add_action( 'pre_get_posts', array( $this, 'filter_list_tables' ) );
+		add_filter( 'views_edit-wsuwp_people_profile', array( $this, 'people_views' ) );
+	}
+
+	/**
+	 * Adds custom roles (called on theme activation).
+	 *
+	 * @since 0.1.0
+	 */
+	public static function add_roles() {
```

Comment:
This can probably be `public function` as it doesn't appeared to be called statically.